### ISSUE TITLE
id-no-dup is falsely reporting duplicates

### ISSUE THREAD

**Issue by nickyblissAviva** [linthtml/linthtml#469]
Since upgrading to 0.9.0 `id-no-dup` it is now reporting thousands of duplicate ids. On investigation many, if not all of the reported duplicate ids are false as it appears to be flagging as a duplicate if the id string is used within an `aria-describedby`  attribute eg.

```
<li class="l-columns__column m-card m-card--link">
	<div class="m-card-content">
		<div class="m-card-content__inner">
			<h3 class="a-heading a-heading--3 a-heading--bold"><a href="/mobile/design-tokens/" class="m-card__link" aria-describedby="carddesc-design-tokens">Design tokens</a></h3>
			<p>Design tokens power the visual attributes of the mobile framework, to maintain a scalable and consistent system for UI development.</p>
		</div>
		<div class="m-card-content__inner m-card-content__inner--bottom">
			<p class="m-card-readmore" id="carddesc-design-tokens" aria-hidden="true"><span>Learn more about design tokens</span></p>
		</div>
	</div>
</li>
```

**Comment by KamiKillertO** [comment]
@nickyblissAviva Are you able to reproduce the issue with only this HTML?
I've tried on my side and no issues are reported for this piece of HTML

**Comment by nickyblissAviva** [comment]
@KamiKillertO Yes it does.
 Even just having something like the following on its own in a file causes the linter to throw an error. If I roll back to 0.8.6 its fine.

```
<h3 id="cards">Cards with heading links</h3>
```

The odd thing is the message it comes back with

```
 2:8  error  The id "cards" is already used at L1353:c60  id-no-dup
```

That test is a 3 line file

**Comment by KamiKillertO** [comment]
🤔 The message is weird.
Looks like the rule is not clearing the id list between files 🤔 .

Can you create a github repo reproducing the issue and post the link here?

**Comment by rishuverkey** [comment]
I am also facing the same issue.

**Comment by KamiKillertO** [comment]
@rishuverkey is there a way for you to share a github repo reproducing the issue?

**Comment by KamiKillertO** [comment]
I'm currently unable to reproduce the issue

**Comment by KamiKillertO** [comment]
@rishuverkey @nickyblissAviva ?

**Comment by nickyblissAviva** [comment]
Will try and fit some time in to create an example repo but unfortunately my job queue has to come first and also work security policies make it impossible to use github so I'll have to set something up on a home machine.

**Comment by KamiKillertO** [comment]
@nickyblissAviva you can send me a link to a gitlab repo or others if it's easier for you. It does not matter to me.
It's just that I'm not able to reproduce on my machine so having an existing repo that I can use for testing.
